### ISSUE TITLE
Make autocomplete match any logically-equivalent package name

### ISSUE THREAD

**Issue by itsayellow** [pipxproject/pipx#555]
**How would this feature be useful?**

Right now autocomplete will only autocomplete if you start with the same exact characters as a venv_dir is named.  However, we allow any package name argument that would match if canonicalized.

We should fix autocomplete to even autocomplete names that don't exactly match the canonicalized internal pipx form, but are logically equivalent in python packaging terms.  e.g. `Fla` could be completed as `Flask` even thought pipx's internal venv directory would be named `flask`.

**Describe the solution you'd like**

Should be as simple as modifying `_autocomplete_list_of_installed_packages()` in `main.py` to accept keyword argument `prefix`, and constructing a list of autocompletes from canonicalized `prefix`.

https://kislyuk.github.io/argcomplete/#specifying-completers

**Describe alternatives you've considered**

Basically the status quo is that autocomplete is more restrictive than pipx in general for package names.
